File: sdkwork-backend-sdk-typescript/src/api/music-generation.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { GenerateMusicForm, PlusApiResultGenerateMusicVO } from '../types';


export class MusicGenerationApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Create music generation task */
  async create(body: GenerateMusicForm): Promise<PlusApiResultGenerateMusicVO> {
    return this.client.post<PlusApiResultGenerateMusicVO>(backendApiPath(`/generation/music/create`), body);
  }

/** Get music generation result */
  async getResult(taskId: string | number): Promise<PlusApiResultGenerateMusicVO> {
    return this.client.get<PlusApiResultGenerateMusicVO>(backendApiPath(`/generation/music/result/${taskId}`));
  }
}

export function createMusicGenerationApi(client: HttpClient): MusicGenerationApi {
  return new MusicGenerationApi(client);
}

File: sdkwork-backend-sdk-go/api/music_generation.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type MusicGenerationApi struct {
    client *sdkhttp.Client
}

func NewMusicGenerationApi(client *sdkhttp.Client) *MusicGenerationApi {
    return &MusicGenerationApi{client: client}
}

// Create music generation task
func (a *MusicGenerationApi) Create(body sdktypes.GenerateMusicForm) (sdktypes.PlusApiResultGenerateMusicVO, error) {
    raw, err := a.client.Post(BackendApiPath("/generation/music/create"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultGenerateMusicVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultGenerateMusicVO](raw)
}

// Get music generation result
func (a *MusicGenerationApi) GetResult(taskId string) (sdktypes.PlusApiResultGenerateMusicVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/generation/music/result/%s", taskId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultGenerateMusicVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultGenerateMusicVO](raw)
}
